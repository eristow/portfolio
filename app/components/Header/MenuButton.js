import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Menu } from '@styled-icons/evaicons-solid/Menu';

const HeaderButton = styled.button.attrs(props => ({
  color: props.color || '#86fcbb',
}))`
  display: inline-flex;
  padding: 0.5em 1em;
  margin: 0.4em 0.4em;
  text-decoration: none;
  border-radius: 4px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: 16px;
  border: 2px solid ${props => props.color};
  color: ${props => props.color};
  background-color: transparent;

  &:active {
    background: ${props => props.color};
    color: white;
  }
`;

HeaderButton.propTypes = {
  color: PropTypes.string,
};

const MenuButton = ({ shouldHaveMarginLeft = false, isOpen, setIsOpen }) => (
  <HeaderButton
    onClick={() => setIsOpen(!isOpen)}
    style={{ marginLeft: shouldHaveMarginLeft ? 'auto' : '' }}
  >
    <Menu size="19" />
  </HeaderButton>
);

MenuButton.propTypes = {
  shouldHaveMarginLeft: PropTypes.bool,
  isOpen: PropTypes.bool,
  setIsOpen: PropTypes.func,
};

export default MenuButton;
