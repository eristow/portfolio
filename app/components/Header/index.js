/**
 *
 * Header
 *
 */

import React from 'react';
import styled from 'styled-components';

import HeaderLink from './HeaderLink';

const NavBar = styled.div`
  text-align: center;
`;

function Header() {
  return (
    <NavBar>
      <HeaderLink to="/">Home</HeaderLink>
      <HeaderLink to="/resume">Resume</HeaderLink>
      <HeaderLink to="/stuff">Stuff</HeaderLink>
      <HeaderLink to="/links">Links</HeaderLink>
    </NavBar>
  );
}

Header.propTypes = {};

export default Header;
