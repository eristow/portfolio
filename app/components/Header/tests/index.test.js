/**
 *
 * Tests for Header
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router/immutable';
import { createMemoryHistory } from 'history';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import configureStore from '../../../configureStore';
import Header from '../index';

describe('<Header />', () => {
  let history;
  let store;

  beforeAll(() => {
    history = createMemoryHistory();
    store = configureStore({}, history);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header />
        </ConnectedRouter>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <Header />
        </ConnectedRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
