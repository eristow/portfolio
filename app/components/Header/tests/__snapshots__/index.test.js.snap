// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render and match the snapshot 1`] = `
<div
  class="Header__NavBar-sc-103fqwf-0 bhratY"
>
  <a
    class="HeaderLink-b4007c-0 cQDFmd"
    color="#bb86fc"
    href="/"
  >
    Home
  </a>
  <a
    class="HeaderLink-b4007c-0 cQDFmd"
    color="#bb86fc"
    href="/resume"
  >
    Resume
  </a>
  <a
    class="HeaderLink-b4007c-0 cQDFmd"
    color="#bb86fc"
    href="/stuff"
  >
    Stuff
  </a>
  <a
    class="HeaderLink-b4007c-0 cQDFmd"
    color="#bb86fc"
    href="/links"
  >
    Links
  </a>
</div>
`;
